### ISSUE TITLE
logstash 1.5 upgrading error -- syntax error, unexpected kNOT

### ISSUE THREAD

**Issue by abcfy2** [elastic/logstash#3352]
I upgrade logstash from 1.4.2 to 1.5.0, here is my config and it works for 1.4.2.
```
input {
    tcp {
        codec => json_lines
        host => "0.0.0.0"
        mode => "server"
        port => "hide_num"
    }
}

filter {
    if ! [tags] or ! [type] {
        drop { }
    }
}

output {
    elasticsearch_http {
        host => "localhost"
        port => "9201"
    }

    if "throttled" not in [tags] and 
        (
            ( [type] == "hawkeyes" and [log_level] == "ERROR" and "Invalid password for user" not in [message]) or
            ( [type] == "rtds" and ( [log_level] == "ERROR" or "Exception" in [message] ))
        ) { 
        email {
           # hide 
        }
    }
}
```

But when I upgrade to 1.5.0, it raised an error:
```
SyntaxError: (eval):58: syntax error, unexpected kNOT
                ( [type] == "hawkeyes" and [log_level] == "ERROR" and "Invalid password for user" not in [message]) or
                                                                                                    ^
        eval at org/jruby/RubyKernel.java:1107
  initialize at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0-java/lib/logstash/pipeline.rb:30
     execute at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0-java/lib/logstash/agent.rb:109
         run at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0-java/lib/logstash/runner.rb:87
        call at org/jruby/RubyProc.java:271
         run at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.0-java/lib/logstash/runner.rb:92
        call at org/jruby/RubyProc.java:271
  initialize at /opt/logstash/vendor/bundle/jruby/1.9/gems/stud-0.0.19/lib/stud/task.rb:12
```

**Comment by duylong** [comment]
I had the same problem. Logstash 1.5.0 don't like condition with multiline.

**Comment by jsvd** [comment]
This has been fixed in #3281 

@abcfy2 Testing with your config:

```
 % bin/logstash -t -f config
The elasticsearch_http output is replaced by the elasticsearch output and will be removed in a future version of Logstash
The following configuration example is based on the options you specified and should work:
elasticsearch {
host => ["localhost"]
port => 9201
codec => "plain"
workers => 1
index => "logstash-%{+YYYY.MM.dd}"
manage_template => true
template_name => "logstash"
template_overwrite => false
flush_size => 100
idle_flush_time => 1
protocol => "http"
}
 {:level=>:warn}
You are using a deprecated config setting "type" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve this same behavior with the new conditionals, like: `if [type] == "sometype" { elasticsearch { ... } }`. If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"type", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn}
You are using a deprecated config setting "tags" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if "sometag" in [tags] { elasticsearch { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"tags", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn}
You are using a deprecated config setting "exclude_tags" set in elasticsearch. Deprecated settings will continue to work, but are scheduled for removal from logstash in the future. You can achieve similar behavior with the new conditionals, like: `if !("sometag" in [tags]) { elasticsearch { ... } }` If you have any questions about this, please visit the #logstash channel on freenode irc. {:name=>"exclude_tags", :plugin=><LogStash::Outputs::ElasticSearch --->, :level=>:warn}
Configuration OK
```

Without the patch it gives the error you posted.

Closing this.
